vm-setup/roles/libvirt/templates/ironic_nodes.json.j2

```diff
@@ -10,15 +10,22 @@
 {
   "nodes": [
   {% for node in vm_nodes %}
+  {% set driver = lookup('env', 'BMC_DRIVER') | default('ipmi') %}
     {
       "name": "{{ node.name|replace('_', '-') }}",
-      "driver": "ipmi",
+      "driver": "{{ driver }}", 
       "resource_class": "baremetal",
       "driver_info": {
-        "ipmi_username": "admin",
-        "ipmi_password": "password",
-        "ipmi_address": "{{ lvars['host_ip'] }}",
-        "ipmi_port": "{{ node.virtualbmc_port }}",
+        "username": "admin",
+        "password": "password",
+        "ip": "{{ lvars['host_ip'] }}",
+        {% if driver =='redfish' -%}
+        "port": "8000",
+        "address": "{{driver}}+{{"http://"}}{{ lvars['host_ip'] }}{{":8000/redfish/v1/Systems/"}}{{vm_id[node.name]}}",
+        {% else -%}
+        "port": "{{ node.virtualbmc_port }}",
+        "address": "{{driver}}{{"://"}}{{lvars['host_ip']}}{{":"}}{{node.virtualbmc_port}}",
```

Comment:
this needs a condition, if the port is defined, otherwise there will be a formatting problem